pkg/services/ngalert/notifier/legacy_storage/receivers.go

```diff
@@ -17,57 +17,57 @@ func (rev *ConfigRevision) DeleteReceiver(uid string) {
 	})
 }
 
-func (rev *ConfigRevision) CreateReceiver(receiver *models.Receiver) error {
+func (rev *ConfigRevision) CreateReceiver(receiver *models.Receiver) (*definitions.PostableApiReceiver, error) {
 	// Check if the receiver already exists.
-	_, err := rev.GetReceiver(receiver.GetUID())
+	_, err := rev.GetReceiver(NameToUid(receiver.Name))
```

Comment:
NOTE: It wasn't clear to me immediately, but `NameToUid` is necessary here over `.GetUid()` because the UID is now empty on creates. Another approach would be to have the `receiver_svc.Create(...)` generate the UID before calling this method, but this is fine as-is.

Reply:
in the future, UID is supposed to be generated by the storage layer, so I decided to keep it in that layer.